hw4.Rmd

```diff
@@ -30,43 +35,98 @@ Before doing anything else, just play with the data to get a sense of it.
 (@) The main outcome of interest is *decision* which records whether the judge granted an appeal or not.  In what proprotion of cases did the judge grant an appeal.
 
 ```{r}
-# [your code here]
+dat.decide.all <- dat %>%
+  group_by(judge) %>%
+  summarise(all = n())
+
+dat.appeal <- dat %>%
+  group_by(judge) %>%
+  filter(decision == "yes") %>%
+  summarise(decision = n())
+
+appeal.table <- full_join(dat.decide.all, dat.appeal, by = "judge") %>%
+  mutate(prop.appeal = decision / all)
+
 ```
 
 (@) There are 12 different judges in the data. A key question is whether different judges have different rates of granting an appeal.  Make a plot showing the number of cases per judge.
 
 ```{r}
-# [your code here]
+#I only count 10 different judges in the data!
+dat %>%
+  group_by(judge) %>%
+  summarise(n())
+
+#Graph:
+graph1 <- ggplot(data = dat.decide.all, 
+                 aes(x = judge, y = all, fill = judge))
+graph1 +  geom_bar(stat = "identity", 
+           colour = "black", 
+           position="dodge",
+           show_guide = FALSE) +
+  xlab("Judge Name") +
+  ylab("Number of Cases") +
+  ggtitle("Number of Cases for Each Judge") +
+  theme_bw() 
 ```
 
 (@) Now plot the rate at which the judges grant appeals. 
 
 ```{r}
-# [your code here]
+graph2 <- ggplot(data = appeal.table, 
+                 aes(x = judge, y = prop.appeal, fill = judge))
+graph2 +  geom_bar(stat = "identity", 
+           colour = "black", 
+           width = 0.7, 
+           position="dodge",
+           show_guide = FALSE) +
+  xlab("Judge Name") +
+  ylab("Proportion of Cases Granted an Appeal") +
+  ggtitle("Proportion of Cases\nGranted an Appeal for each Judge") +
+  theme_bw() 
 ```
 
 (@) Now let's try this with logistic regression because we are going to move to more complicated models.  Fit a logistic regression predicting whetheran appeal will be granted using judge as a categorical predictor. Use Iacobucci as the reference judge to facilitate the comparisons that follow. For more on how to control the reference level of a factor, check out this post on StackOverflow: http://stackoverflow.com/questions/3872070/how-to-force-r-to-use-a-specified-factor-level-as-reference-in-a-regression
 
 ```{r}
-# [your code here]
+#Change reference level:
+levels(dat$judge)
+dat2 <- within(dat, judge <- relevel(judge, ref = "Iacobucci"))
+##check that levels have been adjusted:
+levels(dat2$judge)
+
+#Logistic regression:
+m1 <- glm(decision ~ judge, data = dat2, family = "binomial")
+m1 <- tidy(m1)
+m1
 ```
 
 (@) Using the results of the model calculate the predicted probability that Judge Marceau will grant an appeal.
 
 ```{r}
-# [your code here]
+(exp(m1$estimate[1] + m1$estimate[7])) / (1 + exp(m1$estimate[1] + m1$estimate[7]))
```

Comment:
Try to call the estimates with reference to the judge names instead of numbers because it is more robust and less prone to error. See solutions.